### ISSUE TITLE
feature request: it would be nice if a role could define a (implicit/explicit) tags

### ISSUE THREAD

**Issue by LukaszC** [ansible/ansible#5512]
This issue is similar to the issue #3283, which actually proposed exactly the same feature. I'm creating a new issue to bring attention back to this feature.

It seems useful to be able to run only single task from a playbook. It can be achieved by defining a tag in 'roles' section of the playbook. However this approach has drawback: in every playbook where we will use the role we will have to define the same tag (which may clutter the playbook and introduces duplication).

Implicit tag defined for each of the roles could be a option. Nice I think, since being able to run only selected role(s) seems to be useful. But it can also be considered as little mysterious. Maybe it would be better to introduce '--roles' parameter to ansible-playbook CLI instead.

The other option could be ability to define tags for a role in the role itself (instead of in the playbook). It could be done i.e. by extending role's meta/main.yml file by adding section 'tags' to it (next to 'dependencies') which would define role's tags. The tags defined in meta could sum up with the tags defined in playbook.
This way we could avoid duplication and avoid introducing the implicit tag.

**Comment by bcoca** [comment]
you can add tags to the tasks in the role.

**Comment by mpdehaan** [comment]
Yep, like this even for a shortcut:

```
- include: all_my_tasks.yml
  tags: [  'foo', 'bar' ]
```

But I can see something in the metadata being nice.

**Comment by mpdehaan** [comment]
I'm not really so sure we want this now, because someone could pull in a role from galaxy and be surprised by it messing with tags they might not choose to apply.

**Comment by hkariti** [comment]
That can still happen with the tasks inside the role, so not really a new problem.

**Comment by timruffles** [comment]
this seems like a good feature - people writing galaxy stuff are hopefully more experienced so could namespace their tags. roles feel like good logical levels to turn things on and off

**Comment by roji** [comment]
+1

At the moment I'm manually tagging each and every task all around my playbook. The workaround, of specifying tags in site.yml isn't satisfactory, because roles are reused all around, including via meta.yml dependencies. It seems like there needs to be a _self-contained_ way for a role to define tags.

**Comment by t2d** [comment]
Why not introduce a "tags" list in `meta/main.yml`? As @hkariti  points out, this introduces no new problem, but enables consistent taging _in_ the role.

**Comment by itamaro** [comment]
+1

@mpdehaan - can we get a response about the original reason this request was closed?

as discussed by others already, the claim of "roles pulled from galaxy messing with tags can be surprising" doesn't really hold, because it doesn't change anything from the current situation with tagging individual tasks.

there are many good reasons to tag roles at the role-level. tagging the individual tasks is a workaround, which isn't really practical nor maintainable.
the more reasonable workaround we're implementing is to tag the role at the playbook level, but I find this hacky and dirty, as it separates the tag from the role, and creates duplication if I have multiple playbooks using the same role (and I do).

**Comment by modi-resonai** [comment]
+1

**Comment by hernandesbsousa** [comment]
Is this still a no go? Declaring it only once per role avoids a lot of redundant code.

**Comment by jesuscript** [comment]
+1 would be a great feature, not sure why it's missing

**Comment by hojothum** [comment]
+1

**Comment by reggert** [comment]
:+1:

**Comment by geekofalltrades** [comment]
We currently work around this by having `tasks/main.yml` in our role contain just this:

```
- include: common.yml
  tags: <rolename>
```

And then putting all of the _actual_ role config in `common.yml` and any other necessary files that are included.

It would be great to not have to employ this workaround.

**Comment by Vladimir-csp** [comment]
'tags:' list in meta would be really nice.

**Comment by jkogut** [comment]
:+1:

**Comment by eoli3n** [comment]
+1

**Comment by dwt** [comment]
Can this please be reopened, it would also be really useful for us!

**Comment by DP-M** [comment]
+1 The number of people who use workarounds, and tag every single task say that this is a necessary feature. Thanks!

**Comment by Dem0n3D** [comment]
+1
